File: jenkins_hash.js
function jenkinsHash(str) {
    for (var hash = 0, i = 0, len = str.length; i < len; ++i) {
        hash += (str.charCodeAt(i) >>> 0);
        hash = hash >>> 0
        hash += ((hash << 10) >>> 0);
        hash = hash >>> 0
        hash ^= ((hash >>> 6) >>> 0);
        hash = hash >>> 0
    }
    hash += ((hash << 3) >>> 0);
    hash = hash >>> 0
    hash ^= ((hash >>> 11) >>> 0);
    hash = hash >>> 0
    hash += ((hash << 15) >>> 0);
    hash = hash >>> 0
    return hash >>> 0;
}

File: jenkins_hash.rb
def jenkins_hash(str)
  MAX_32_BIT = 4294967295
  hash = 0
  str.length.times { |n|
    hash += str[n].ord
    hash &= MAX_32_BIT
    hash += ((hash << 10) & MAX_32_BIT)
    hash &= MAX_32_BIT
    hash ^= hash >> 6
  }
  hash += (hash << 3 & MAX_32_BIT)
  hash &= MAX_32_BIT
  hash ^= hash >> 11
  hash += (hash << 15 & MAX_32_BIT)
  hash &= MAX_32_BIT
  hash
end
